mysql-test/suite/sys_vars/t/innodb_default_encryption_key_id_basic.test

```diff
@@ -1,3 +1,62 @@
---skip Not implemented
-
 SELECT @@global.innodb_default_encryption_key_id;
+
+--let $error_log=$MYSQL_TMP_DIR/my_restart.err
+
+--replace_result $KEYRING_PLUGIN KEYRING_PLUGIN $MYSQL_TMP_DIR MYSQL_TMP_DIR
+--let $restart_parameters=restart:--early-plugin-load="keyring_file=$KEYRING_PLUGIN" --keyring_file_data=$MYSQL_TMP_DIR/keyring/keyring --default-table-encryption=ONLINE_TO_KEYRING --innodb-encryption-rotate-key-age=15 --innodb-encryption-threads=4 --innodb-default-encryption-key-id=4294967295 --log-error=$error_log
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR
+--source include/restart_mysqld.inc
+
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_FILE=$error_log
+--let SEARCH_PATTERN=\[Server\] option 'innodb-default-encryption-key-id': unsigned value 4294967295 adjusted to 4294967294\.
+--source include/search_pattern_in_file.inc
+
+--let $restart_parameters=restart:--early-plugin-load="keyring_file=$KEYRING_PLUGIN" --keyring_file_data=$MYSQL_TMP_DIR/keyring/keyring --default-table-encryption=ONLINE_TO_KEYRING --innodb-encryption-rotate-key-age=15 --innodb-encryption-threads=4 --innodb-default-encryption-key-id=-1 --log-error=$error_log
+--replace_result $MYSQL_TMP_DIR MYSQL_TMP_DIR
+--source include/restart_mysqld.inc
+
+--let ABORT_ON=NOT_FOUND
+--let SEARCH_FILE=$error_log
+--let SEARCH_PATTERN=\[Server\] option 'innodb-default-encryption-key-id': value -1 adjusted to 0\.
+--source include/search_pattern_in_file.inc
+
+SET innodb_default_encryption_key_id=4294967295;
+SET innodb_default_encryption_key_id=-1;
+
+SET innodb_default_encryption_key_id=0;
+
+--error ER_ENCRYPTION_KEY_ID_VALUE_OUT_OF_RANGE
+CREATE TABLE t_exceeded_id (a varchar(255)) ENCRYPTION_KEY_ID=4294967295;
+
+--error ER_PARSE_ERROR
+CREATE TABLE t_below_0 (a varchar(255)) ENCRYPTION_KEY_ID=-1;
+
+CREATE TABLE t_MAX (a varchar(255)) ENCRYPTION_KEY_ID=4294967294;
+CREATE TABLE t_MIN (a varchar(255)) ENCRYPTION_KEY_ID=0;
+
+--error ER_ENCRYPTION_KEY_ID_VALUE_OUT_OF_RANGE
+ALTER TABLE t_MAX ENCRYPTION_KEY_ID=4294967295;
+--error ER_PARSE_ERROR
+ALTER TABLE t_MIN ENCRYPTION_KEY_ID=-1;
```

Comment:
Why don't we get the same error code for both cases?

Reply:
@satya-bodapati, min and max value checks are done in different places in this case. For max value we directly check in parser rules which we can change and use whichever error we want. The min value check is done by parser itself based on datatype assigned to the symbol which is unsigned in this case - "ENCRYPTION_KEY_ID_SYM opt_equal real_ulong_num". We could in general change datatype for ENCRYPTION_KEY_ID_SYM to signed and do check its min value manually. Do you think we should follow this way?

Reply:
Got it! thanks